include/phasar/PhasarLLVM/Utils/AnalysisPrinterBase.h

```diff
@@ -0,0 +1,44 @@
+#ifndef PHASAR_PHASARLLVM_UTILS_ANALYSISPRINTERBASE_H
+#define PHASAR_PHASARLLVM_UTILS_ANALYSISPRINTERBASE_H
+
+#include "llvm/Support/raw_ostream.h"
+
+namespace psr {
+
+template <typename AnalysisDomainTy> struct Warning {
+  using n_t = typename AnalysisDomainTy::n_t;
+  using d_t = typename AnalysisDomainTy::d_t;
+  using l_t = typename AnalysisDomainTy::l_t;
+
+  n_t Instr;
+  d_t Fact;
+  l_t LatticeElement;
+
+  // Constructor
+  Warning(n_t Inst, d_t DfFact, l_t Lattice)
+      : Instr(std::move(Inst)), Fact(std::move(DfFact)),
+        LatticeElement(std::move(Lattice)) {}
+};
+
+template <typename AnalysisDomainTy> struct Results {
```

Comment:
I wonder whether the name Results is a bit too generic. It describes results of a dataflow analysis, but we also have PT, TH, CG etc. which cannot be expressed with this one.